### PR TITLE
Add missing multiset class to the standard library bindings

### PR DESCRIPTION
**Pull request by da-woods**
It's largely just copied from set because it has basically
the same interface.

### CODE REVIEW

File: Cython/Includes/libcpp/set.pxd [hunk 1]
```diff
@@ -59,3 +59,61 @@ cdef extern from "<set>" namespace "std" nogil:
         iterator upper_bound(const T&)
         const_iterator const_upper_bound "upper_bound"(const T&)
         #value_compare value_comp()
+
+    cdef cppclass multiset[T]:
+        ctypedef T value_type
+        cppclass iterator:
```
Comment (scoder) [inline Cython/Includes/libcpp/set.pxd:65]:
```suggestion
        ctypedef T value_type

        cppclass iterator:
```

File: Cython/Includes/libcpp/set.pxd [hunk 2]
```diff
@@ -59,3 +59,61 @@ cdef extern from "<set>" namespace "std" nogil:
...
+            iterator operator++()
+            iterator operator--()
+            bint operator==(reverse_iterator)
+            bint operator!=(reverse_iterator)
+        cppclass const_iterator(iterator):
+            pass
+        cppclass const_reverse_iterator(reverse_iterator):
+            pass
+        multiset() except +
+        multiset(multiset&) except +
```
Comment (scoder) [inline Cython/Includes/libcpp/set.pxd:82]:
```suggestion
        cppclass const_reverse_iterator(reverse_iterator):
            pass

        multiset() except +
        multiset(multiset&) except +
```

File: Cython/Includes/libcpp/unordered_set.pxd [hunk 3]
```diff
@@ -45,19 +40,71 @@ cdef extern from "<unordered_set>" namespace "std" nogil:
...
+        void rehash(size_t)
+        void reserve(size_t)
+        size_t bucket_count()
+        size_t max_bucket_count()
+        size_t bucket_size(size_t)
+        size_t bucket(const T&)
+
+    cdef cppclass unordered_multiset[T,HASH=*,PRED=*,ALLOCATOR=*]:
+        ctypedef T value_type
+        cppclass iterator:
```
Comment (scoder) [inline Cython/Includes/libcpp/unordered_set.pxd:63]:
```suggestion
        ctypedef T value_type

        cppclass iterator:
```

File: Cython/Includes/libcpp/unordered_set.pxd [hunk 4]
```diff
@@ -45,19 +40,71 @@ cdef extern from "<unordered_set>" namespace "std" nogil:
...
+            iterator operator++()
+            iterator operator--()
+            bint operator==(reverse_iterator)
+            bint operator!=(reverse_iterator)
+        cppclass const_iterator(iterator):
+            pass
+        cppclass const_reverse_iterator(reverse_iterator):
+            pass
+        unordered_multiset() except +
+        unordered_multiset(unordered_multiset&) except +
```
Comment (scoder) [inline Cython/Includes/libcpp/unordered_set.pxd:80]:
```suggestion
        cppclass const_reverse_iterator(reverse_iterator):
            pass

        unordered_multiset() except +
        unordered_multiset(unordered_multiset&) except +
```

**Review by scoder — COMMENTED** [review]
We should grant innocent readers at least a bit of formatting.

File: tests/run/cpp_stl_multiset.pyx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    for v in vals:
+        ms.insert(v)
+    ms.erase(to_remove)
+    return sorted([ item for item in ms ])
+
+def test_unordered_multiset_find_erase(vals, to_remove):
+    """
+    >>> test_multiset_find_erase([1,2,2,3, -1], 1)
+    [-1, 2, 2, 3]
+    >>> test_multiset_find_erase([1,2,2,3, -1], 2)  # removes a single copy of 2
```
Comment (da-woods) [inline tests/run/cpp_stl_multiset.pyx:98]:
```suggestion
    >>> test_unordered_multiset_find_erase([1,2,2,3, -1], 1)
    [-1, 2, 2, 3]
    >>> test_unordered_multiset_find_erase([1,2,2,3, -1], 2)  # removes a single copy of 2
```

**Pull request merged**